src/aspire/basis/dirac.py

```diff
@@ -1,5 +1,79 @@
+import logging
+import numpy as np
+
 from aspire.basis import Basis
+from aspire.utils.matrix import roll_dim, unroll_dim
+from aspire.utils.matlab_compat import m_flatten, m_reshape
+
+logger = logging.getLogger(__name__)
 
 
 class DiracBasis(Basis):
-    pass
+    """
+    Define a derived class for Dirac basis
+    """
+    def __init__(self, sz, mask=None):
+        """
+        initialize an object for Dirac Basis
+        :param sz: The size of the vectors for which to define the basis.
+        :param mask: A boolean mask of size sz indicating which coordinates
+            to include in the basis (default true(sz)).
+        """
+        if mask is None:
```

Comment:
It does call superclass. Not sure there is any addition.

Reply:
It's generally best to call it first thing in the constructor so it doesn't overwrite anything.

Reply:
For other basis classes, the base constructor is designed to be called at the end of the derived constructor because the `_build` function is included in the base constructor. I will follow the consistent way if there is no specific better reason.

Reply:
Do you remember why we did that? It seems like it could cause problems.

Reply:
If the base constructor is called first, that  `_build` will be called before the parameters specified in the derived class are initialized. This is not what we want, we want all parameters are initialized first then call  `_build` to initailize more complex data structure.